r-package/grf/vignettes/categorical_inputs.Rmd

```diff
@@ -0,0 +1,92 @@
+---
+title: "Categorical inputs"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{categorical_inputs}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+set.seed(123)
+```
+
+In this example, we will illustrate several ways of dealing with categorical variables when using grf.
+
+```{r setup}
+library(grf)
+```
+
+One of the approaches below relies on `grf`'s sister package [`sufrep`](https://github.com/grf-labs/sufrep). Let's install and load it first.
+```r
+devtools::install_github("grf-labs/sufrep")
+library(sufrep)
+```
+
+Let's pretend we would like to estimate mileage per gallon (`mpg`) from number of cylinders (`cyl`),
+quarter-mile time (`qsec`), and car brand name (`brand`, created below).
+
+```{r dataset}
+# Create a categorical column with brand name
+df <- within(mtcars, {
+  # E.g. 'Mazda RX4' --> 'Mazda'
+  brand <- factor(sapply(rownames(mtcars), function(x) strsplit(x, " ")[[1]][1]))
+})
+
+x <- c("cyl", "qsec") # Continuous variables
+g <- c("brand")       # Categorical variable
+
+head(df[c(x, g)])
+```
+
+This code would raise an error, because data is not numerical.
+
+```{r problem}
+# rf <- regression_forest(X=df[c(x, g)], Y=df$mpg)
```

Comment:
nothing wrong here but just noticed `validate_X` being inconsistent: doing as.matrix(df[c(x, g)]) causes it to provide the error message instead of the call to the bindings

Reply:
And maybe we should not expose that inconsistency here? (i.e fix it in another pr)

Reply:
Maybe this error message could even be updated to refer to sufrep: https://github.com/grf-labs/grf/blob/master/r-package/grf/R/input_utilities.R#L8

Reply:
Yeah, I noticed it as well. I'll leave the error message as is for now, will fix it later.